backend/src/modules/user/user.controller.ts

```diff
@@ -0,0 +1,71 @@
+import {Body, Controller, Get, HttpStatus, Post, Res, Param, UseGuards } from '@nestjs/common';
+import { UserService } from './user.service';
+import { ApiBearerAuth, ApiOperation, ApiResponse } from '@nestjs/swagger';
+import { UserEntity } from './user.entity';
+import { CreateUserDto } from './dto/create-user.dto';
+import { Response} from 'express';
+import { AuthGuard } from '@nestjs/passport';
+import { AppError } from '../../common/error/AppError';
+import { AppErrorTypeEnum } from '../../common/error/AppErrorTypeEnum';
+
+@Controller('user')
+export class UserController {
+    constructor(private readonly userService: UserService) {}
+
+    /** Get all endpoint - authorization required */
+    @Get('')
+    @UseGuards(AuthGuard('bearer'))
+    @ApiOperation({title: 'Get List of All Users'})
+    //For this request an existing user should try to login first and then authorize with 'Bearer TOKEN',
+    //where TOKEN - is the token recieved by logging in.
+    @ApiBearerAuth()
+    @ApiResponse({ status: 200, description: 'User found.'})
+    @ApiResponse({ status: 401, description: 'User not authorized for this request.'})
+    @ApiResponse({ status: 404, description: 'No Users found.'})
+    public async getAllUsers(@Res() res: Response) {
+        console.log('Get to /user | getAllUsers');
```

Comment:
Būtų pretty cool naudot [nest utils logginimui](https://docs.nestjs.com/techniques/logger), o ne console.log

Reply:
Čia docsai prastai paaiškina kaip jį naudot, tai [yra SO klausimas](https://stackoverflow.com/questions/52650895/how-to-use-nestjs-logging-service)